Invoices / Providers section

ferblape opened the pull request:
Closes #1191
Closes #739
Closes #1314 
Closes #1315  

### What does this PR do?

This PR implements a new section for Invoices and Providers in the GobiertoBudgets module.

- This section uses data from PopulateData
- This section needs to be enabled in the configuration of GobiertoBudgets

### How should this be manually tested?

It's enabled in a coule of sites:

- http://getafe.gobify.net/presupuestos/proveedores-facturas

- http://mataro.gobify.net/presupuestos/proveedores-facturas

### Improvements worth to consider

1. ~Add a spinner while the data is being loaded~

2. ~Synchronize DC.js filters with data table, therefore:~
  - ~when a filter is applied in a DC chart the table updates~
  - ~when the table is filtering by a row the charts are updated~

3. Extract `overlay` as a component.

Crashillo committed:
Headings invoices/providers

Crashillo committed:
Layout approach

Crashillo committed:
Chart by month

Data similar

Crashillo committed:
Additional notes

Crashillo committed:
DC charts displayed

Crashillo committed:
Deselected elements style

Crashillo committed:
d3.v3 Integration mixed

Crashillo committed:
Format data axis/labels

Crashillo committed:
Table filter

Crashillo committed:
Charts styled

Crashillo committed:
File by request & table data formatting

Crashillo committed:
I18n

Crashillo committed:
filters

Crashillo committed:
Data filtered & freelance calculations

Crashillo committed:
Adjusts

Crashillo committed:
Empty values

Crashillo committed:
Axis positions & translations

Crashillo committed:
Texts calculations

Crashillo committed:
New calculations

Crashillo committed:
Chart titles

Crashillo committed:
Fix locale

Commit:
Merge branch 'master' into 1191-providers

ferblape commented:
Markup complete, starting with the integration.

Commit:
Integrate providers and invoices

Commit:
Test ES6 support in staging

Commit:
Enable ES6 syntax in production

Commit:
Amend key name

Commit:
Merge branch 'master' into 1191-providers

ferblape commented:
@furilo please review, and give your opinion about the suggested improvements.

Commit:
Add missing locale

Commit:
Add providers updated information

Commit:
Allow sites without place linked

ferblape commented:
@Crashillo look the graphs in Safari and Chrome:

![screen shot 2018-01-09 at 18 28 36](https://user-images.githubusercontent.com/17616/34734129-fe26d03e-f56a-11e7-9109-0683bb38e8b0.png)

Crashillo committed:
Force width based on parent

Webkit does not calculate elastic axis

furilo commented:
I'd just review 

- [x] the % of companies/individuals (we need to flip it?)
- [x] "El 0 % de las facturas concentran el 10 % de la facturación" - maybe something is wrong with this calculation?

In order to close this PR. I've created #1315 and #1314 - I'll create other issues for small things.

Crashillo committed:
fix bug after filter #1314

Crashillo committed:
date filter improved

Crashillo committed:
all page elements refreshed when crossfilter changes

Crashillo committed:
clean code

Crashillo committed:
add spinner

Crashillo commented on app/assets/stylesheets/comp-graph-dc.scss:
```diff
@@ -64,3 +64,33 @@
 }
+
+//Overlays
+.overlay-container {
```
This could be turn into a component (partial) to be used all around

furilo replied:
Yes, go ahead.

Crashillo committed:
table filters chart also

Crashillo committed:
bug in filters

Crashillo committed:
fix bug table filters

Crashillo committed:
Removed debug stuff

Crashillo committed:
overlay as component

ferblape commented:
Please, merge `master`, because there's a conflict in ` app/assets/stylesheets/application.scss`

Crashillo committed:
Merge branch 'master' into 1191-providers

furilo committed:
Adjusting a little bit the spinner overlay

furilo committed:
Removing insight that was not showing interesting info

Commit:
Fix I18n terms for mean and median invoices amounts

Commit:
Fix freelance/company calculations

Commit:
Calculate then percentile 10

Commit:
Show more digits

ferblape commented:
I have fixed:

-   the % of companies/individuals (we need to flip it?) > it was wrong, I have flipped it
-  "El 0 % de las facturas concentran el 10 % de la facturación" - maybe something is wrong with this calculation? > the calculation is correct, I have added decimal digits

furilo commented:
Should we merge this and deploy?

Crashillo commented:
There are several issues related. Shall we merge now and solve them later (different PRs)?
